devolutions-gateway/src/recording/muxer.rs

```diff
@@ -0,0 +1,43 @@
+use std::{
+    path::{Path, PathBuf},
+    sync::Arc,
+    time::Duration,
+};
+
+use anyhow::Context;
+
+pub async fn remux(input: impl AsRef<Path>) -> anyhow::Result<()> {
+    let path = Arc::new(input.as_ref().to_owned());
+    // Could potentially be a very slow operation
+    tokio::task::spawn_blocking(|| remux_impl(path)).await??;
+    Ok(())
+}
+
+fn remux_impl(input: Arc<PathBuf>) -> anyhow::Result<()> {
```

Comment:
suggestion: You can take a `PathBuf` by move instead of using `Arc`. There is no need for shared ownership.